### ISSUE TITLE
When a building wall is destroyed it creates a "ledge" area leading into solid rock

### ISSUE THREAD

**Issue by tsorrells** [CleverRaven/Cataclysm-DDA#35266]
# Describe the bug

I've noticed recently that any time a building wall has been destroyed whether by zombies, vehicle collision, or sledgehammer that it doesn't just crumble, but produces a "ledge" area in something like a 3x3 pattern around the wall and stepping into it drops the player into solid rock with no way out.

# Steps To Reproduce

1. Destroy a building wall with any suitable means: sledgehammer, vehicle collision, explosives, etc.
2. Notice ground around destroyed wall is a "ledge" and effectively impassable

# Expected behavior

Destroying a wall should simply lead to it being reduced to a pile of debris or rubble of some kind and should not produce ledges or any kind of pit into the ground.

# Screenshots

![CDDA Ledges](https://user-images.githubusercontent.com/1437138/68077340-f7241b00-fd8f-11e9-8510-447a2f9ff208.png)

# Versions and configuration

Version: 0.D-9059-gae829af (tiles)
Build: 9853 - experimental

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.D-9059-gae829af [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Filthy Clothing [filthy_morale],
    Simplified Nutrition [novitamins]

**Comment by KurzedMetal** [comment]
I saw this happen with trees too. Seems like everything leaves a "hole" in the wrong now.

**Comment by tsorrells** [comment]
Oh you're right.  I tried crashing into a tree and it did indeed turn it into a ledge area.  I don't think I did that before, so I don't know what used to happen.

I've attached a pic of the ledge areas that resulted after ramming my car into 2 of them.  And then for comparison, I chopped down a tree with a toolbox and it fell over normally.
![CDDA Ledges 2](https://user-images.githubusercontent.com/1437138/68080992-8ce19980-fdd4-11e9-8eae-b85aafb68406.png)

**Comment by everythingisonfire** [comment]
I'm having this issue on build 9859 as well.

<img width="960" alt="cataclysm-tiles_MORpkik8pH" src="https://user-images.githubusercontent.com/56804684/68182962-06aa8c00-ff69-11e9-9daf-ec6fd6efc670.png">

**Comment by Sons-of-Man** [comment]
Can also confirm this. Happening too any tree or wall that is destroyed it seems. I don't recall it happening on 9847 which is what I updated from.

**Comment by Gilferedir** [comment]
I think I may have figured this one out, but I hardly know what I'm doing so I might have messed something else up as it took me forever to figure out how to even compile the game, so I don't know if I am comfortable submitting a pull request. I went through the releases between 9847 and 9859 and found a change to the map.h file in PR (#35218) Fix clang tidy errors. I reverted the change `        
 834       void batter( const tripoint &p, int power, int tries = 1, bool silent = false );` back to `834    void batter( const tripoint &p, int power, int tries = 1, const bool silent = false );` and the version I compiled seems to be working now.  If it doesn't get fixed soon, I might try my hand at a PR but I haven't tested it very extensively yet.

**Comment by Sons-of-Man** [comment]
So far my temporary fix that I was told by Vorm is turning z levels back on, with it off this issue arises. Did just that and all the holes for the most part were auto filled with rock flooring and walls and trees are acting normal again. Really strange.

**Comment by Gilferedir** [comment]
Yes. After some testing with mods and menu settings it seems it is a Z level thing and was able to reproduce it even with the change I made.

**Comment by tsorrells** [comment]
@Sons-of-Man Ok cool, at least we have a workaround now.  Can you add z-levels to a world that was already generated with them disabled?

@Gilferedir Ok thanks for updating us.

**Comment by Sons-of-Man** [comment]
@tsorrells You can, just need to restart your game after turning them back on for it to take affect.

**Comment by Gilferedir** [comment]
I tracked it all the way back to Release 9794. No idea how to fix it though. It seems someone made some changes to how collapsing works.

**Comment by DaviBones** [comment]
Good work @Gilferedir, I just confirmed this was indeed introduced in bb41c00, I am taking a look at the code now and will submit a PR soon.
